cmd/options/options.go

```diff
@@ -47,41 +44,132 @@ type NodeProblemDetectorOptions struct {
 	// ServerAddress is the address to bind the node problem detector server.
 	ServerAddress string
 
+	// exporter options
+
+	// k8sExporter options
+	// EnableK8sExporter is the flag determining whether to report to Kubernetes.
+	EnableK8sExporter bool
+	// ApiServerOverride is the custom URI used to connect to Kubernetes ApiServer.
+	ApiServerOverride string
+
+	// prometheusExporter options
+	// PrometheusServerPort is the port to bind the Prometheus scrape endpoint. Use 0 to disable.
+	PrometheusServerPort int
+	// PrometheusServerAddress is the address to bind the Prometheus scrape endpoint.
+	PrometheusServerAddress string
+
+	// problem daemon options
+
+	// SystemLogMonitorConfigPaths specifies the list of paths to system log monitor configuration
+	// files.
+	// SystemLogMonitorConfigPaths is used by the deprecated option --system-log-monitors. The new
+	// option --config.system-log-monitor will stored the config file paths in MonitorConfigPaths.
+	SystemLogMonitorConfigPaths []string
+	// CustomPluginMonitorConfigPaths specifies the list of paths to custom plugin monitor configuration
+	// files.
+	// CustomPluginMonitorConfigPaths is used by the deprecated option --custom-plugin-monitors. The
+	// new option --config.custom-plugin-monitor will stored the config file paths in MonitorConfigPaths.
+	CustomPluginMonitorConfigPaths []string
+	// MonitorConfigPaths specifies the list of paths to configuration files for each monitor.
+	MonitorConfigPaths types.ProblemDaemonConfigPathMap
+
 	// application options
 
 	// NodeName is the node name used to communicate with Kubernetes ApiServer.
 	NodeName string
 }
 
 func NewNodeProblemDetectorOptions() *NodeProblemDetectorOptions {
-	return &NodeProblemDetectorOptions{}
+	return &NodeProblemDetectorOptions{MonitorConfigPaths: types.ProblemDaemonConfigPathMap{}}
 }
 
 // AddFlags adds node problem detector command line options to pflag.
 func (npdo *NodeProblemDetectorOptions) AddFlags(fs *pflag.FlagSet) {
 	fs.StringSliceVar(&npdo.SystemLogMonitorConfigPaths, "system-log-monitors",
```

Comment:
MarkDeprecated?

Reply:
Sorry, what do you mean by "MarkDeprecated?" i.e. add a comment? Or is there some good procedure we could follow?
Currently I'm documenting that it's deprecated in the help text: `This option is deprecated, replaced by --config.system-log-monitor, and will be removed. NPD will panic if both --system-log-monitors and --config.system-log-monitor are set.`

Reply:
Changed to use MarkDeprecated to handle this logic.
Now, when user use the deprecated option, a warning will be printed as this:

> Flag --system-log-monitors has been deprecated, replaced by --config.system-log-monitor. NPD will panic if both --system-log-monitors and --config.system-log-monitor are set.